Automattic/jetpack
# AI Assistant: convert HTML to markdown in the request content

This PR converts the content provided by blocks (usually enriched via HTML) to markdown before being used to perform the request to get generated content. It fixes the issue that, sometimes, the response contains plain HTML.

**This PR introduces the new [turndown](https://github.com/mixmark-io/turndown) package dependency.**

Fixes https://github.com/Automattic/jetpack/issues/30739

## Proposed changes:

* AI Assistant: convert HTML to markdown in the request content

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Create some content using paragraph blocks with formated context (bold, italic, etc)
* Create an AI Assistant block instance
* Generate content based on the post (`Continue writing`, for instance)
* Confirm that the content in the prompt contains markdown format

before | after
-------|------
<img width="406" alt="Screenshot 2023-05-15 at 15 35 19" src="https://github.com/Automattic/jetpack/assets/77539/404a384d-6aee-4d78-a11f-bade2a457ac3">|<img width="360" alt="Screenshot 2023-05-15 at 15 34 13" src="https://github.com/Automattic/jetpack/assets/77539/030956a9-4025-4c90-bb1a-0f85fe08f400">
